src/commands/database/create.mjs

```diff
@@ -5,23 +5,47 @@ import { FaunaError } from "fauna";
 import { container } from "../../cli.mjs";
 import { throwForError } from "../../lib/fauna.mjs";
 import { formatObjectForShell } from "../../lib/misc.mjs";
+import { getSecret, retryInvalidCredsOnce } from "../../lib/fauna-client.mjs";
 
-async function createDatabase(argv) {
-  const logger = container.resolve("logger");
+function validate(argv) {
+  if (!argv.secret && !argv.database) {
+    throw new Error(
+      "No secret or database provided. Please use either --secret or --database.",
+    );
+  }
+  return true;
+}
+
+async function runCreateQuery(secret, argv) {
   const { fql } = container.resolve("fauna");
   const { runQuery } = container.resolve("faunaClientV10");
-
-  try {
-    await runQuery({
-      url: argv.url,
-      secret: argv.secret,
-      query: fql`Database.create({
+  return runQuery({
+    secret,
+    url: argv.url,
+    query: fql`
+      Database.create({
         name: ${argv.name},
         protected: ${argv.protected ?? null},
         typechecked: ${argv.typechecked ?? null},
         priority: ${argv.priority ?? null},
       })`,
-    });
+  });
+}
+
+async function createDatabase(argv) {
+  const secret = await getSecret();
+  const logger = container.resolve("logger");
+
+  try {
+    if (argv.secret === secret) {
+      // If we are using a user provided secret, we should not
+      // try to refresh it if it is bad.
+      await runCreateQuery(secret, argv);
+    } else {
+      await retryInvalidCredsOnce(secret, async (secret) =>
+        runCreateQuery(secret, argv),
+      );
+    }
```

Comment:
My assumption was this behavior is already handled via: https://github.com/fauna/fauna-shell/blob/5d621c949d4841bd6d36349dba9474e2f184ee60/src/lib/auth/databaseKeys.mjs#L76-L80

Did it not work that way in your testing?

Reply:
Ahh cool I didn't know this was already baked in